usdt-impl/src/no-linker.rs

```diff
@@ -130,9 +130,10 @@ fn extract_probe_records_from_section() -> Result<Section, crate::Error> {
     // writable (to implement one-time registration), so an immutable variable here
     // leads to _two_ sections, one writable and one read-only. A mutable variable
     // here ensures this ends up in a mutable section, the same as the probe records.
-    #[cfg(target_os = "illumos")]
+    #[cfg(any(target_os = "illumos", target_os = "freebsd"))]
     #[link_section = "set_dtrace_probes"]
```

Comment:
@ahl I am open to using `dtrace_probes`, but changing the section name to "dtrace_probes" for FreeBSD only will introduce some messy `#[cfg(*)]` in various places since we also need to change any occurrence of `__(start|stop)_$section` as well.

Reply:
> the `set_` prefix is an idiosyncratic illumos-ism. Unless FreeBSD does the same, I'd suggest just `"dtrace_probes".

The illumos aspect where `set_` is prefixed to the linker set in the `<sys/linker_set.h>` header came from the FreeBSD implementation where the same thing happens (from a somewhat older sys/sys/linker_set.h from the FreeBSD tree):

```
 73 #define __MAKE_SET_QV(set, sym, qv)                     \
 74         __WEAK(__CONCAT(__start_set_,set));             \
 75         __WEAK(__CONCAT(__stop_set_,set));              \
 76         static void const * qv                          \
 77         __NOASAN                                        \
 78         __set_##set##_sym_##sym __section("set_" #set)  \
 79         __used = &(sym)
```

Therefore I expect one may need the same section naming logic.

Reply:
@ahl @rmustacc Thank you for much for the information, it is way beyond my knowledge otherwise. I never stop being fascinated by UNIX history and how it is affecting us. 

I can confirm the same macro is still in the FreeBSD tree and is being used in the kernel by the sysinit framework. I however do not know if the `set_` prefix is being used elsewhere. I personally lean toward keeping it as with the `set_` prefix, so the patch can be less invasive and easier to maintain.